R/assert_is_unique_dt.R

```diff
@@ -0,0 +1,47 @@
+#' Are the rows of the input data.table unique?
+#'
+#' Checks to see if the rows of the input data.table are unique for each
+#' combination of the id columns.
+#'
+#' @param dt input data.table to check.
+#' @param id_cols character vector of id columns that uniquely identify each row
+#' of `dt`.
+#' @inheritParams assertive.types::assert_is_data.table
+#'
+#' @return
+#' * `is_unique_dt` returns a logical.
+#' * `assert_is_unique_dt` returns nothing but throws an error if `is_unique_dt`
+#' returns FALSE.
+#'
+#' @export
+#'
+#' @examples
+#' input_dt <- data.table::data.table(location = "France", year = 2010,
+#'                                    sex = "female",
+#'                                    age_start = 0:95,
+#'                                    value1 = 2, value2 = 4)
+#' id_cols <- c("location", "year", "sex", "age_start")
+#' is_unique_dt(input_dt, id_cols)
+#' assert_is_unique_dt(input_dt, id_cols)
+
```

Comment:
For consistency we should not leave a space between documentation a function

Reply:
What about the extra #' ?

Reply:
I think in some of the other functions I was leaving `#'`. Do we want that or just go directly to function definition?

Reply:
For now I removed the line between the documentation and function definition.